### PR TITLE
Various Loot Pile and Dungeon objects Stage Uno

### PR DESCRIPTION
**Pull request by Mechoid**
Any lootpile can now have a list of

Assorted Exosuit lootpiles for PoIs and Events.

Random spawners for future loot piles or PoIs.

Log stack spawners, and a sif-wood spawner.

So. Old. So Many god damn update commits, please squash-merge if it's accepted.

### CODE REVIEW

**Comment by Arokha** [comment]
Not to be that guy who pops out of the woodwork to tell you a better way of doing things, but, you can rebase instead of merge to avoid those commits

**Comment by mistyLuminescence** [comment]
Also, not sure if you fixed it with this, but I tried putting a random energy gun in a lootpile spawner earlier, and it spawned the "random energy gun" item, not an actual random gun - it'd probably be the same for all random items, unless this PR fixes that.

File: code/game/objects/structures/loot_piles.dm [hunk 1]
```diff
@@ -568,3 +567,267 @@ Loot piles can be depleted, if loot_depleted is turned on.  Note that players wh
...
+		/obj/item/mecha_parts/mecha_equipment/weapon/energy/riggedlaser,
+		/obj/item/mecha_parts/mecha_equipment/tool/hydraulic_clamp,
+		/obj/item/mecha_parts/mecha_equipment/tool/drill,
+		/obj/item/mecha_parts/mecha_equipment/generator
+		)
+
+	rare_loot = list(
+		/obj/item/mecha_parts/mecha_equipment/weapon/energy/laser,
+		/obj/item/mecha_parts/mecha_equipment/generator/nuclear,
+		/obj/item/mecha_parts/mecha_equipment/tool/jetpack
```
Comment (Belsima) [inline code/game/objects/structures/loot_piles.dm]: i am proud

**Comment by Leshana** [comment]
Sorry, I didn't see this PR while it was open.    Instead of `spawn()` in new and deleting self, you can use `initialize()`. and `INITIALIZE_HINT_QDEL`

**Pull request merged**